spec/controllers/courses_controller_spec.rb

```diff
@@ -1,6 +1,85 @@
 describe CoursesController do
   let(:instructor) { create :instructor }
   let(:course) { instructor.course }
+
+  describe 'role switching methods' do
+    let(:end_user) { create :end_user }
+    let(:subject) do
+      post_as instructor, :switch_role, params: { 'id' => course.id, 'effective_user_login' => end_user.user_name }
```

Comment:
In general use symbol keys, not strings. So `id: course.id` instead of `'id' => course.id`, for example.